src/libraries/System.Security.Cryptography.Encoding/tests/Cbor.Tests/CborReaderTests.Tag.cs

```diff
@@ -0,0 +1,349 @@
+            InvalidOperationException exn = Assert.Throws<InvalidOperationException>(() => reader.ReadTag());
+
+            Assert.Equal("Data item major type mismatch.", exn.Message);
+        }
+
+        [Fact]
+        public static void ReadTag_NestedTagWithMissingPayload_ShouldThrowFormatException()
+        {
+            byte[] data = "9fc2ff".HexToByteArray();
+            var reader = new CborReader(data);
+
+            reader.ReadStartArray();
+            reader.ReadTag();
+            Assert.Equal(CborReaderState.FormatError, reader.Peek());
+            Assert.Throws<FormatException>(() => reader.ReadEndArray());
+        }
+
+        [Theory]
+        [InlineData("8201c202")] // definite length array
+        [InlineData("9f01c202ff")] // equivalent indefinite-length array
+        public static void ReadTag_CallingEndReadArrayPrematurely_ShouldThrowInvalidOperationException(string hexEncoding)
+        {
+            // encoding is valid CBOR, so should not throw FormatException
+            byte[] data = hexEncoding.HexToByteArray();
+            var reader = new CborReader(data);
+
+            reader.ReadStartArray();
+            reader.ReadInt64();
+            reader.ReadTag();
+            Assert.Equal(CborReaderState.UnsignedInteger, reader.Peek());
+            Assert.Throws<InvalidOperationException>(() => reader.ReadEndArray());
+        }
+
+        [Theory]
+        [InlineData("a102c202")] // definite length map
+        [InlineData("bf02c202ff")] // equivalent indefinite-length map
+        public static void ReadTag_CallingEndReadMapPrematurely_ShouldThrowInvalidOperationException(string hexEncoding)
+        {
+            // encoding is valid CBOR, so should not throw FormatException
+            byte[] data = hexEncoding.HexToByteArray();
+            var reader = new CborReader(data);
+
+            reader.ReadStartMap();
+            reader.ReadInt64();
+            reader.ReadTag();
+            Assert.Equal(CborReaderState.UnsignedInteger, reader.Peek());
+            Assert.Throws<InvalidOperationException>(() => reader.ReadEndArray());
+        }
+
+        [Theory]
+        [InlineData("2013-03-21T20:04:00Z", "c074323031332d30332d32315432303a30343a30305a")]
+        [InlineData("2013-03-21T20:04:00Z", "c11a514b67b0")]
+        [InlineData("2013-03-21T20:04:00.5Z", "c1fb41d452d9ec200000")]
+        [InlineData("2020-04-09T14:31:21.3535941+01:00", "c07821323032302d30342d30395431343a33313a32312e333533353934312b30313a3030")]
+        [InlineData("2020-04-09T11:41:19.12-08:00", "c0781c323032302d30342d30395431313a34313a31392e31322d30383a3030")]
+        [InlineData("2020-04-09T13:31:21Z", "c11a5e8f23a9")]
+        [InlineData("1970-01-01T00:00:00Z", "c100")]
+        [InlineData("1969-12-31T23:59:59Z", "c120")]
+        [InlineData("1960-01-01T00:00:00Z", "c13a12cff77f")]
+        public static void ReadDateTimeOffset_SingleValue_HappyPath(string expectedValueString, string hexEncoding)
```

Comment:
Since I can't read CBOR the way I can DER: Are any of these using indefinite strings for the date?  I assume there should be some that do... (if it's verboten, negative tests, otherwise, this positive test).

Reply:
The spec is vague in that regard. Here it is in its entirety:
```
Tag value 0 is for date/time strings that follow the standard format described in [RFC3339], as refined by Section 3.3 of [RFC4287].
```
I'd therefore assume they should be accepted, provided the conformance level for the reader permits it.

Reply:
Addressed.